packages/store/addon/-private/system/schema-definition-service.ts

```diff
@@ -83,15 +72,15 @@ export class DSModelSchemaDefinitionService {
  * @param normalizedModelName already normalized modelName
  * @return {*}
  */
-export function getModelFactory(store: CoreStore, cache, normalizedModelName: string): Model | null {
+export function getModelFactory(store: CoreStore, cache, normalizedModelName: string): RecordInstance | null {
   let factory = cache[normalizedModelName];
 
   if (!factory) {
     factory = _lookupModelFactory(store, normalizedModelName);
 
     if (!factory) {
       //Support looking up mixins as base types for polymorphic relationships
-      factory = _modelForMixin(store, normalizedModelName);
+      factory = store._modelForMixin(normalizedModelName);
```

Comment:
instead of adding a private method to store and injecting it from the model package, we should use the `HAS_MODEL_PACKAGE`  build flag to conditionally import a private function from that package.

```ts
let _modelForMixin;
if (HAS_MODEL_PACKAGE) {
 _modelForMixin = require('@ember-data/model/-private')._modelForMixin;
}

// ....

if (HAS_MODEL_PACKAGE) {
  factory = _modelForMixin(store, normalizedModelName);
}
```

possibly with a module scope cache for the method to avoid looking it up each time.